src/Data/Vector/NonEmpty.hs

```diff
@@ -2646,3 +2649,17 @@ iscanr f b = NonEmptyVector . V.iscanr f b . _neVec
 iscanr' :: (Int -> a -> b -> b) -> b -> NonEmptyVector a -> NonEmptyVector b
 iscanr' f b = NonEmptyVector . V.iscanr' f b . _neVec
 {-# INLINE iscanr' #-}
+
+-- | \(\mathcal{O}(n)\). The 'intersperse' function takes an element and a NonEmptyVector
```

Comment:
It looks like you copypasted something like the implementation for `intersperse` from `Data.List.NonEmpty`, which isn't a very good implementation for something like `vector` or `nonempty-vector` for several reasons: 

1. This function is not `O(n)`: `cons` for `Vector` is `O(n)`, and vector consing for `n-1`-many elements is actually `O(n^2)`. This makes the implementation rather unsatisfactory, and in fact, worse than the just using a list for very large vectors. 

2. There is an issue out to resolve the suboptimal vector `cons`-based implementation (posted here: https://github.com/haskell/vector/issues/421), which discusses the potential streams-based implementation that I would welcome in this package when the time comes for that work to be done in `vector`. Otherwise, we'd be importing a very suboptimal implementation into this package, breaking from `vector`'s API. 

3. I think there's benefit in matching the `vector` API, since it makes this package a near drop-in replacement for `vector`, minus some fiddling and pattern matching to enforce the nonempty invariants in one's code. We'd be deviating from that until point #2 is implemented.

So i have to reject this on those grounds.

Reply:
At least, in its current incarnation. If you want to help the `vector` package move the work along for `intersperse`, I'm happy to keep this open and accept it with a deference to the `vector` 
implementation!